src/components/DatePicker.js

```diff
@@ -0,0 +1,462 @@
+    if (this.props.title) {
+      return (
+        <div key='title' style={styles.title}>
+          {this.props.title}
+        </div>
+      );
+    }
+  }
+
+  _toggleCalendar () {
+    this.setState({
+      showCalendar: !this.state.showCalendar
+    });
+  }
+
+  _handleBlur () {
+    this.setState({
+      showCalendar: false
+    });
+  }
+
+  _renderCaret () {
+    if (this.state.showCalendar) {
+      return (
+        <Icon
+          onClick={this._toggleCalendar.bind(this)}
+          size='20px'
+          style={[styles.navIcon, styles.navRight]}
+          type='caret-up'
+        />);
+    } else {
+      return (
+        <Icon
+          onClick={this._toggleCalendar.bind(this)}
+          size='20px'
+          style={[styles.navIcon, styles.navRight]}
+          type='caret-down'
+        />);
+    }
+  }
+
+  render () {
+    const selectedDate = moment.unix(this.state.selectedDate).locale(this.props.locale);
+    const currentDate = this.state.currentDate ? this.state.currentDate.locale(this.props.locale) : selectedDate;
+
+    return (
+      <div
+        onBlur={this._handleBlur.bind(this)}
+        style={[styles.component, { fontSize: this.props.fontSize }, styles.clearFix]}
+        tabIndex='0'
+      >
+        <div key='componentTop' style={[styles.componentTop, this.state.showCalendar && styles.componentTopOpen]}>
+          {this._renderSelectedDate()}
+          {this.props.showCaret ? this._renderCaret() : null}
+        </div>
+        <div key='componentBottom' style={[styles.componentBottom, this.state.showCalendar && styles.calendarShow]}>
+          {this._renderTitle(styles)}
+          <div key='calendarHeader' style={[styles.calendarHeader, { borderBottomStyle: this.props.showDayBorders ? 'solid' : 'none' }, styles.clearFix]}>
+            <Icon
+              onClick={this._handlePreviousClick.bind(this)}
+              size='32px'
+              style={[styles.navIcon, styles.navLeft, this.props.showDayBorders && styles.borderRight]}
+              type='caret-left'
+            />
+            {currentDate.format('MMMM YYYY')}
+            <Icon
+              onClick={this._handleNextClick.bind(this)}
+              size='32px'
+              style={[styles.navIcon, styles.navRight, this.props.showDayBorders && styles.borderLeft]}
+              type='caret-right'
+            />
+          </div>
+          <div style={styles.calendarContainer}>
+            {this._renderMonthTable(currentDate, selectedDate)}
+          </div>
+          <div style={styles.clearFix}></div>
+        </div>
+        {this._renderScrim(styles)}
+      </div>
+    );
+  }
+}
+
+DatePicker.propTypes = {
+  closeOnDateSelect: React.PropTypes.bool,
+  fontSize: React.PropTypes.string,
+  format: React.PropTypes.string,
+  locale: React.PropTypes.string,
+  minimumDate: React.PropTypes.string,
+  onDateSelect: React.PropTypes.func,
+  scrimStyle: React.PropTypes.object,
+  selectedDate: React.PropTypes.string,
```

Comment:
This should be named `defaultDate` since once a user starts interacting with the date picker we stop using it. It will also help us differentiate between the `selectedDate` parameter on the state.